src/postamble.js

```diff
@@ -188,19 +188,11 @@ Module['callMain'] = function callMain(args) {
   ensureInitRuntime();
 
   var argc = args.length+1;
-  function pad() {
-    for (var i = 0; i < {{{ QUANTUM_SIZE }}}-1; i++) {
-      argv.push(0);
-    }
-  }
-  var argv = [allocate(intArrayFromString(Module['thisProgram']), 'i8', ALLOC_NORMAL) ];
-  pad();
-  for (var i = 0; i < argc-1; i = i + 1) {
-    argv.push(allocate(intArrayFromString(args[i]), 'i8', ALLOC_NORMAL));
-    pad();
+  var argv = _malloc(argc * {{{ Runtime.POINTER_SIZE }}});
+  HEAP32[argv >> 2] = allocateUTF8(Module['thisProgram']);
+  for (var i = 0; i < argc-1; i++) {
+    HEAP32[(argv >> 2) + 1 + i] = allocateUTF8(args[i]);
```

Comment:
Can it be:

```js
for (var i = 1; i < argc; i++) { 
  HEAP32[(argv >> 2) + i] = allocateUTF8(args[i - 1]);
}
```

? Just two characters less, though.

Reply:
Thanks, done. I think it's clearer your way.